api/support/api.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package support contains the support service API client.
+package support
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/juju/errors"
+	"github.com/juju/romulus/wireformat/support"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+	"gopkg.in/macaroon.v1"
+)
+
+var DefaultURL = "https://api.jujucharms.com/omnibus/v2"
+
+// SupportAuthClient defines the interface available to clients of the support api.
+type SupportAuthClient interface {
+	// Authorize returns the support macaroon for the specified model
+	Authorize(modelUUID, supportLevel string) (*macaroon.Macaroon, error)
+}
+
+var _ SupportAuthClient = (*client)(nil)
+
+type httpClient interface {
+	// Do sends the given HTTP request and returns its response.
+	Do(*http.Request) (*http.Response, error)
+	// DoWithBody is like Do except that the given body is used
+	// for the body of the HTTP request, and reset to its start
+	// by seeking if the request is retried. It is an error if
+	// req.Body is non-zero.
+	DoWithBody(req *http.Request, body io.ReadSeeker) (*http.Response, error)
+}
+
+// client is the implementation of the Client interface.
+type client struct {
+	client  httpClient
+	baseURL string
+}
+
+// ClientOption defines a function which configures a Client.
+type ClientOption func(h *client) error
+
+// HTTPClient returns a function that sets the http client used by the API
+// (e.g. if we want to use TLS).
+func HTTPClient(c httpClient) func(h *client) error {
+	return func(h *client) error {
+		h.client = c
+		return nil
+	}
+}
+
+// BaseURL sets the base url for the api client.
+func BaseURL(url string) func(h *client) error {
+	return func(h *client) error {
+		h.baseURL = url
+		return nil
+	}
+}
+
+// NewSupportAuthClient returns a new support authorization client.
+func NewSupportAuthClient(options ...ClientOption) (SupportAuthClient, error) {
```

Comment:
why do we have NewSupportAuthClient and NewClient? perhaps the NewClient does not need to be exported?